pyccel/codegen/printing/ccode.py

```diff
@@ -813,7 +816,11 @@ def _print_FunctionDef(self, expr):
 
         if len(expr.results) > 1:
             self._additional_args.append(expr.results)
+        if self._target_language == 'c':
+            self._allocs = [i for i in expr.local_vars if i.allocatable and i.is_ndarray]
```

Comment:
Can you explain why `if self._target_language == 'c':` is needed here?

Reply:
because the cwrapper is writing the frees of the garbage collector I think I can remove this with what u requested of just overriding the free_allocs method